### PR TITLE
[TE] Add query for anomaly functions under an application

### PR DESCRIPTION
**Pull request by cyy0714**
Implement the query method in AnomalyFuncitonDAO for future application summary and evaluation.

### CODE REVIEW

**Comment by cyenjung** [comment]
The schema in this PR is actually correct. The column application in alert_config_index was missing in the create-schema.sql.

**Comment by apucher** [comment]
looks like there's quite a few nulls in the index for application in my copy. we'll have to address this

File: thirdeye/thirdeye-pinot/src/test/resources/schema/create-schema.sql [hunk 1]
```diff
@@ -160,12 +160,14 @@ create index override_config_base_id_idx ON override_config_index(base_id);
...
     active boolean,
     name varchar(500) not null,
+    application VARCHAR(500) not null,
     base_id bigint(20) not null,
     create_time timestamp,
     update_time timestamp default current_timestamp,
     version int(10),
     CONSTRAINT uc_alert_name UNIQUE (name)
 ) ENGINE=InnoDB;
+create index alert_confix_application_idx on alert_config_index(application);
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/test/resources/schema/create-schema.sql]: "confix" typos

Comment (cyy0714) [reply thirdeye/thirdeye-pinot/src/test/resources/schema/create-schema.sql]: Fixed! thanks for the comment

**Pull request merged**